runners/core-java/src/main/java/org/apache/beam/runners/core/ReduceFnRunner.java

```diff
@@ -598,26 +598,14 @@ private void processElement(Map<W, W> windowToMergeResult, WindowedValue<InputT>
       nonEmptyPanes.recordContent(renamedContext.state());
 
       // Make sure we've scheduled the end-of-window or garbage collection timer for this window.
-      Instant timer = scheduleEndOfWindowOrGarbageCollectionTimer(directContext);
+      scheduleGarbageCollectionTimer(directContext);
 
       // Hold back progress of the output watermark until we have processed the pane this
       // element will be included within. If the element is too late for that, place a hold at
       // the end-of-window or garbage collection time to allow empty panes to contribute elements
       // which won't be dropped due to lateness by a following computation (assuming the following
       // computation uses the same allowed lateness value...)
-      @Nullable Instant hold = watermarkHold.addHolds(renamedContext);
-
-      if (hold != null) {
```

Comment:
We no longer want to confirm that the timer/hold pairing is sensible? Should we instead tighten this (if hold is <= EOW, then timer should be <= EOW, if hold is <= GC time, timer should be > EOW and <= GC time?)

Reply:
The prior check was an attempt to make sure the if we got an EOW hold then we got an EOW timer, versus when we get a GC hold we got a GC timer. But two issues: (a) the bug I'm fixing is where we get an EOW hold but a GC timer, and this didn't catch it and (b) now we just always get a GC timer while the hold could be at an early timestamp (if based on elements) or GC time, and there's no error case.